httptestutil/dump.go

```diff
@@ -0,0 +1,77 @@
+package httptestutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"os"
+
+	"github.com/felixge/httpsnoop"
+)
+
+// DumpTo wraps an http.Handler in a new handler; \the new handler dumps requests and responses to a writer, using the httputil.DumpRequest and httputil.DumpResponse functions
+func DumpTo(handler http.Handler, writer io.Writer) http.Handler {
+	// use the same default as http.Server
+	if handler == nil {
+		handler = http.DefaultServeMux
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		dump, err := httputil.DumpRequest(r, true)
+		if err != nil {
+			_, _ = fmt.Fprintf(writer, "error dumping request: %#v", err)
```

Comment:
If you don't want anything from the func return, you can just leave that out completely , e.g here (and a few other places in this func:
```suggestion
			fmt.Fprintf(writer, "error dumping request: %#v", err)
```